packages/libs/pactjs/package.json

```diff
@@ -24,7 +24,7 @@
     }
   ],
   "main": "lib/index.js",
-  "types": "lib/index.d.ts",
+  "types": "dist/pactjs.d.ts",
```

Comment:
This one causes trouble:

```
[typescript] Encountered 2 TypeScript issues:
[typescript] Error: src/tests/mkCap.test.ts:54:57 - (TS2339) Property 'toPactInteger' does not exist on type 'PactNumber'.
[typescript] Error: src/tests/mkCap.test.ts:55:62 - (TS2339) Property 'toPactDecimal' does not exist on type 'PactNumber'.
 ---- Compile encountered an error (1577ms) ---- 
-------------------- Finished (1.756s) --------------------
Encountered 2 errors:
  [typescript] src/tests/mkCap.test.ts:54:57 - (TS2339) Property 'toPactInteger' does not exist on type 'PactNumber'.
  [typescript] src/tests/mkCap.test.ts:55:62 - (TS2339) Property 'toPactDecimal' does not exist on type 'PactNumber'.
```

Reply:
The issue is probably caused by the triple slash directive in `PactNumber.ts` that ends up in `lib/PactNumber.d.ts`, but not in `dist/pactjs.d.ts`.

Reply:
API Extractor seems to not handle the triple slash directive:

```ts
/// <reference path="./extend-bignumber.ts" />
```

It's just gone (I think it should inline `extend-bignumber.ts`). To me this seems normal and valid usage of TypeScript. AE seems to [support only `types` and `lib` references](https://github.com/microsoft/rushstack/blob/main/apps/api-extractor/src/generators/DtsRollupGenerator.ts#L95-L102).

I did try a few things with `files` in `tsconfig.json` and `bundledPackages` in `api-extractor.json`, but to no avail.

Not proud of it, but this does seem to do the trick:

```
"build": "heft build --clean && cat src/extend-bignumber.ts >> dist/pactjs.d.ts"
```

There are other ways around this, but making exceptions to the `types": "dist/[pkg].d.ts"` rule or other TypeScript hacks to me sound even worse.

Any ideas welcome, though!

If we don't come up with a real fix and merge this I could open an issue in the rushstack repo.

Reply:
Unfortunately not, declaration gets stripped, not sure how to work around this.